brilliant-builders/firestarter/game_engine/utils.py

```diff
@@ -0,0 +1,9 @@
+from typing import Set
+
+
+def get_all_subclasses(cls: type) -> Set[type]:
+    """Return all subclasses of a give cls class."""
+    return set(
+        cls.__subclasses__()).union(
+            [s for c in cls.__subclasses__() for s in get_all_subclasses(c)]
+    )
```

Comment:
Not really a fan of this function. You could have made it more readable by using a for loop:
```python
def get_all_subclasses(cls: type) -> Set[type]:
    """Return all subclasses of a given class."""
    subclasses = set()
    for subclass in cls.__subclasses__():
        subclasses.add(subclass)
        subclasses.update(get_all_subclasses(cls))
    return subclasses
```